utilities.asm

```diff
@@ -9,6 +9,7 @@
 ;--------------------------------------------------------------------------------
 GetSpriteID:
 	JSR AttemptItemSubstitution
+	PHX
```

Comment:
I think you're missing a PLX at line 28 (early return for bottles).

Reply:
You may be right, this might be the cause of the bottle related crashes in DoorRando